### PR TITLE
Compile navigation row via own method

### PR DESCRIPTION
**Pull request by rabauss**
Compile navigation row via own method, to get the possibility to overwrite/extend it in own extensions.

### CODE REVIEW

**Comment by aschempp** [comment]
The PR generally looks good to me. I just dislike the whole navigation module concept because
 a. it is inside the parent `Module` class
 b. it does not use the `knp/menu-bundle`
 c. its hell of ugly code…

@rabauss why are you using these methods anyway? Are you overriding the navigation module or creating a second one?

**Comment by rabauss** [comment]
The Code is really a ugly.
But in one project we used or overwritten this module. Finally, we could have written the module ourselves.
I just wanted to give the oportunity to do easier overwrites in the future ;-)

**Comment by Toflar** [comment]
I‘m fine with that change. Doesn‘t do any harm😊

**Review by ausi — DISMISSED** [review]

**Comment by leofeyer** [comment]
It seems this PR needs some rebasing. 🙈

**Comment by leofeyer** [comment]
Contao 4.9 is now in the review phase in which we try to review and merge the remaining pull requests that have been created during the development phase. Your pull request does not yet meet the requirements to be reviewed though:

* Make sure that the PR is feature complete.
* Rebase the PR onto the latest master branch.
* Address the requested changes if any.
* Make sure that all CI checks are successful.

Please fix this soon if you can, because the review phase ends on January 15th and we need some time to review the PR before we can merge it.

File: core-bundle/src/Resources/contao/modules/Module.php [hunk 1]
```diff
@@ -351,89 +351,104 @@ protected function renderNavigation($pid, $level=1, $host=null, $language=null)
...
+	 * Compile the navigation row and return it as array
+	 *
+	 * @param PageModel $objPage
+	 * @param PageModel $objSubpage
+	 * @param string    $subitems
+	 * @param string    $href
+	 *
+	 * @return array
+	 */
+	protected function compileNavigationRow($objPage, $objSubpage, $subitems, $href)
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/modules/Module.php]:
```suggestion
	protected function compileNavigationRow(PageModel $objPage, PageModel $objSubpage, $subitems, $href)
```

**Review by leofeyer — CHANGES_REQUESTED** [review]

**Review by leofeyer — APPROVED** [review]

**Review by Toflar — APPROVED** [review]

File: core-bundle/src/Resources/contao/modules/Module.php (hunk 1 above)
Comment (rabauss) [reply core-bundle/src/Resources/contao/modules/Module.php]: When creating the PR I was not used to the fact that TypeHints can also be used in legacy code ;-)
Should we also add the `string` hint for the last parameters, shouldn't we?

**Comment by leofeyer** [comment]
Thank you @rabauss.

File: core-bundle/src/Resources/contao/modules/Module.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/Module.php]: No, we only use type hints that were already supported in PHP 5.6 in the legacy code.

**Pull request merged**